### PR TITLE
dev-libs/spdlog: bump 1.4.2

### PR DESCRIPTION
**Pull request by lovesegfault**
dev-libs/spdlog: bump 1.4.2

Package-Manager: Portage-2.3.76, Repoman-2.3.17
Signed-off-by: Bernardo Meurer <bernardo@standard.ai>

### CODE REVIEW

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 1]
```diff
@@ -18,23 +18,20 @@ fi
 
 LICENSE="MIT"
 SLOT="0/1"
-IUSE="test"
+IUSE="test static-libs"
+RESTRICT="!test? ( test )"
 
 DEPEND="
 	>=dev-libs/libfmt-5.0.0
```
Comment (stkw0) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: this version of spdlog works with libfmt 5? If i'm not wrong this should be changed to dev-libs/libfmt:0/6, same for live ebuild.

thanks :)

Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: I opened an upstream issue asking exactly this, I couldn't find anything in the source that led me to believe it _wouldn't_ work with 5.x

https://github.com/gabime/spdlog/issues/1264

**Comment by stkw0** [comment]
btw we should attach the bug https://bugs.gentoo.org/696710 to this PR

File: dev-libs/spdlog/spdlog-1.4.2.ebuild (hunk 1 above)
Comment (stkw0) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: oh, nice. Tomorrow I will try to take a look

**Comment by lovesegfault** [comment]
Done @stkw0 :)

**Comment by stkw0** [comment]
Thanks.

I noticed two more things. It seems that by default benchmarks are not
compiled but examples are. We should disable it. Also tests are failing for
me but only if executed with meson.

File: dev-libs/spdlog/spdlog-1.4.2.ebuild (hunk 1 above)
Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: The author confirmed it as `>=dev-libs/libfmt-5.3.0`. I've set it to that

**Comment by lovesegfault** [comment]
@stkw0 All fixed up, I reverted back to CMake at the author's recommendation. Tests and everything are working.

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 2]
```diff
@@ -10,29 +10,31 @@ HOMEPAGE="https://github.com/gabime/spdlog"
...
 DEPEND="
-	>=dev-libs/libfmt-5.0.0
+	>=dev-libs/libfmt-5.3.0
 "
 RDEPEND="${DEPEND}"
 
-PATCHES=( "${FILESDIR}/${PN}-1.0.0-unbundle-fmt.patch" )
-
 src_configure() {
-	rm -r include/spdlog/fmt/bundled || die
```
Comment (stkw0) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: shouldn't we removed the bundled fmtlib btw?

Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: I don't think so, they don't end up in the image if you force the use of an external fmt, which we do.

Comment (a17r) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: well, at least this way you make sure it does not build against bogus headers by accident

Comment (stkw0) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: nice. The last day I saw bundled files being installed on the image, but definitely was my mistake :P

Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: I'll reintroduce in `src_prepare` and see how it goes

Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: Worked well, fixed :)

**Comment by lovesegfault** [comment]
Is this good to go @stkw0?

**Comment by stkw0** [comment]
> Is this good to go @stkw0?

yes, at least I have nothing else to say.

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+DESCRIPTION="Very fast, header only, C++ logging library"
+HOMEPAGE="https://github.com/gabime/spdlog"
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/gabime/${PN}"
+else
+	SRC_URI="https://github.com/gabime/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="amd64"
```
Comment (a17r) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: you're adding a new ebuild with stable keywords? repoman didn't complain?

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+BDEPEND="
+	virtual/pkgconfig
+"
+DEPEND="
+	>=dev-libs/libfmt-5.3.0
+"
+RDEPEND="${DEPEND}"
+
+src_prepare() {
+	default
```
Comment (a17r) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: don't run default, put cmake-utils_src_prepare here instead.

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+LICENSE="MIT"
+SLOT="0/1"
+IUSE="static-libs test"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	virtual/pkgconfig
+"
+DEPEND="
+	>=dev-libs/libfmt-5.3.0
```
Comment (a17r) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: please check if you need to add a slot operator here.

File: dev-libs/spdlog/spdlog-1.4.2.ebuild (hunk 3 above)
Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: Oh wow, that's a travesty. I have no idea how this got through repoman :thinking:

File: dev-libs/spdlog/spdlog-1.4.2.ebuild (hunk 5 above)
Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: Yep, was missing subslot

**Comment by a17r** [comment]
From what I can see it is still installing to `/usr/lib64/spdlog/cmake/` instead of `/usr/lib64/cmake/spdlog/`?

**Comment by stkw0** [comment]
> During merge:
> 
> ```
> <<<          obj /usr/lib64/cmake/spdlog/spdlogTargets.cmake
> <<<          obj /usr/lib64/cmake/spdlog/spdlogConfigVersion.cmake
> <<<          obj /usr/lib64/cmake/spdlog/spdlogConfig.cmake
> >>> /usr/lib64/spdlog/cmake/
> >>> /usr/lib64/spdlog/cmake/spdlogConfigTargets.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfigTargets-gentoo.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfig.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfigVersion.cmake
> ```
> 
> That's not going to work.

This was an upstream bug (already fixed: see https://github.com/gabime/spdlog/commit/4858d7e454e66d347024a074749535b91ccbeed9). We can apply that patch as is.

btw @lovesegfault  could you also attach https://bugs.gentoo.org/699346 ? Thank you.

**Comment by stkw0** [comment]
> During merge:
> 
> ```
> <<<          obj /usr/lib64/cmake/spdlog/spdlogTargets.cmake
> <<<          obj /usr/lib64/cmake/spdlog/spdlogConfigVersion.cmake
> <<<          obj /usr/lib64/cmake/spdlog/spdlogConfig.cmake
> >>> /usr/lib64/spdlog/cmake/
> >>> /usr/lib64/spdlog/cmake/spdlogConfigTargets.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfigTargets-gentoo.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfig.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfigVersion.cmake
> ```
> 
> That's not going to work.

This was an upstream bug (already fixed: see https://github.com/gabime/spdlog/commit/4858d7e454e66d347024a074749535b91ccbeed9). We can apply that patch as is.

btw @lovesegfault  could you also attach https://bugs.gentoo.org/699346 ? Thank you.

**Comment by stkw0** [comment]
> During merge:
> 
> ```
> <<<          obj /usr/lib64/cmake/spdlog/spdlogTargets.cmake
> <<<          obj /usr/lib64/cmake/spdlog/spdlogConfigVersion.cmake
> <<<          obj /usr/lib64/cmake/spdlog/spdlogConfig.cmake
> >>> /usr/lib64/spdlog/cmake/
> >>> /usr/lib64/spdlog/cmake/spdlogConfigTargets.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfigTargets-gentoo.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfig.cmake
> >>> /usr/lib64/spdlog/cmake/spdlogConfigVersion.cmake
> ```
> 
> That's not going to work.

This was an upstream bug (already fixed: see https://github.com/gabime/spdlog/commit/4858d7e454e66d347024a074749535b91ccbeed9). We can apply that patch as is.

btw @lovesegfault  could you also attach https://bugs.gentoo.org/699346 ? Thank you.

**Comment by lovesegfault** [comment]
I'm out sick, will be looking at this soon.

**Comment by stkw0** [comment]
> I'm out sick, will be looking at this soon.

Thank you very much for the initial work. Hope your health is better.


@a17r  I've created another PR with the patch for the 1.4.2 version, see #14113

**Comment by a17r** [comment]
Closing obsolete PR after cb4209494b05520fa84511c0ec3df825085af73e, thanks for your work.

**Pull request closed without merging**